File: app/src/main/java/com/kakyiretechnologies/darktheme/java/SettingsFragment.java
package com.kakyiretechnologies.darktheme.java;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatDelegate;
import androidx.preference.ListPreference;
import androidx.preference.PreferenceFragmentCompat;

import com.kakyiretechnologies.darktheme.R;

public class SettingsFragment extends PreferenceFragmentCompat {

    private final String light = "light";
    private final String dark = "dark";
    private final String themeKey = "theme"; //assign the same key value in the preferences xml file

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.preferences, rootKey);

        //find the key of the theme preference
        ListPreference theme = findPreference(themeKey);

        //listen to events or change by user to apply theme
        theme.setOnPreferenceChangeListener((preference, newValue) -> {
            changeTheme(newValue.toString());
            return true;
        });
    }

    private void changeTheme(String value) {

        if (value.equals(light)) {
            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
            return;
        }

        if (value.equals(dark)) {
            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
            return;
        }

        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
    }


}

File: app/src/main/java/com/kakyiretechnologies/darktheme/kotlin/SettingsFragment.kt
package com.kakyiretechnologies.darktheme.kotlin

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatDelegate
import androidx.preference.ListPreference
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import com.kakyiretechnologies.darktheme.R

class SettingsFragment : PreferenceFragmentCompat() {

    private val light = "light"
    private val dark = "dark"
    private val themeKey = "theme"//assign the same key value in the preferences xml file

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.preferences, rootKey)

        //find the key of the theme preference
        val theme = findPreference<ListPreference>(themeKey)
       
        //listen to events or change by user to apply theme
        theme?.onPreferenceChangeListener =
            Preference.OnPreferenceChangeListener { preference, newValue ->
                
                changeTheme(newValue.toString())
                Log.d(TAG, "onCreatePreferences: $newValue")
                
                true
            }
    }

    private fun changeTheme(value: String) = when (value) {
        light -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
        dark -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
        else -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
    }

}
private const val TAG = "SettingsFragment"
